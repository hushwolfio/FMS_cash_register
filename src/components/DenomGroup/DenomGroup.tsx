import type { FC } from 'react';
import type { Denominations } from '../../types';
import { DenomAdjuster } from '../DenomAdjuster';
import { Flex } from '@mantine/core';

interface DenomGroupProps {
    denominations: Denominations;
    setDenomAmount: (denomination: string, count: string | number) => void;
}

export const DenomGroup: FC<DenomGroupProps> = ({ denominations, setDenomAmount }) => (
    <Flex gap="md" direction="row" wrap="wrap" justify="center">
        {Object.keys(denominations).map((denom) => (
            <DenomAdjuster
                value={denominations[denom]}
                key={denom}
                denom={denom}
                setDenomAmount={setDenomAmount}
            />
        ))}
    </Flex>
);