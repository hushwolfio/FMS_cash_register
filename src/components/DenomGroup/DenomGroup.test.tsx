import '../../matchMedia';
import { render } from '@testing-library/react';
import { describe, it, expect, vi } from 'vitest';
import { DenomGroup } from './DenomGroup';
import { MantineProvider } from '@mantine/core';

describe('DenomGroup', () => {
    const denominations = { '1': 1, '5': 1, '10': 1, '20': 1, '50': 1, '100': 1 };
    const setDenomAmountSpy = vi.fn();
    it('should render', () => {
        const { container } = render(<MantineProvider><DenomGroup denominations={denominations} setDenomAmount={setDenomAmountSpy} /></MantineProvider>);
        expect(container).toMatchSnapshot();
    });

    it('should render with denominations', () => {
        const { getByText } = render(<MantineProvider><DenomGroup denominations={denominations} setDenomAmount={setDenomAmountSpy} /></MantineProvider>);
        expect(getByText('Add/Remove $1')).toBeInTheDocument();
    });
});