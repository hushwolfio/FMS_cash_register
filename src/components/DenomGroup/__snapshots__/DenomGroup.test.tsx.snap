// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DenomGroup > should render 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    class="m_8bffd616 mantine-Flex-root __m__-r0"
    style="gap: var(--mantine-spacing-md); justify-content: center; flex-wrap: wrap; flex-direction: row;"
  >
    <div
      class="m_e2f5cd4e m_46b77525 mantine-InputWrapper-root mantine-NumberInput-root"
      style="width: 120px;"
    >
      <label
        class="m_8fdc1311 mantine-InputWrapper-label mantine-NumberInput-label"
        for="mantine-bf3keclke"
        id="mantine-bf3keclke-label"
      >
        Add/Remove $1
      </label>
      <div
        class="m_6c018570 mantine-Input-wrapper mantine-NumberInput-wrapper"
        data-variant="default"
        data-with-right-section="true"
        style="--input-right-section-width: var(--ni-right-section-width-sm);"
      >
        <input
          aria-invalid="false"
          class="m_8fb7ebe7 mantine-Input-input mantine-NumberInput-input"
          data-variant="default"
          id="mantine-bf3keclke"
          inputmode="numeric"
          type="text"
          value="1"
        />
        <div
          class="m_82577fc2 mantine-Input-section mantine-NumberInput-section"
          data-position="right"
        >
          <div
            class="m_95e17d22 mantine-NumberInput-controls"
          >
            <button
              aria-hidden="true"
              class="mantine-focus-auto m_80b4b171 mantine-NumberInput-control m_87cf2631 mantine-UnstyledButton-root"
              data-direction="up"
              tabindex="-1"
              type="button"
            >
              <svg
                fill="none"
                style="transform: rotate(180deg);"
                viewBox="0 0 15 15"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M3.13523 6.15803C3.3241 5.95657 3.64052 5.94637 3.84197 6.13523L7.5 9.56464L11.158 6.13523C11.3595 5.94637 11.6759 5.95657 11.8648 6.15803C12.0536 6.35949 12.0434 6.67591 11.842 6.86477L7.84197 10.6148C7.64964 10.7951 7.35036 10.7951 7.15803 10.6148L3.15803 6.86477C2.95657 6.67591 2.94637 6.35949 3.13523 6.15803Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
            </button>
            <button
              aria-hidden="true"
              class="mantine-focus-auto m_80b4b171 mantine-NumberInput-control m_87cf2631 mantine-UnstyledButton-root"
              data-direction="down"
              tabindex="-1"
              type="button"
            >
              <svg
                fill="none"
                viewBox="0 0 15 15"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M3.13523 6.15803C3.3241 5.95657 3.64052 5.94637 3.84197 6.13523L7.5 9.56464L11.158 6.13523C11.3595 5.94637 11.6759 5.95657 11.8648 6.15803C12.0536 6.35949 12.0434 6.67591 11.842 6.86477L7.84197 10.6148C7.64964 10.7951 7.35036 10.7951 7.15803 10.6148L3.15803 6.86477C2.95657 6.67591 2.94637 6.35949 3.13523 6.15803Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="m_e2f5cd4e m_46b77525 mantine-InputWrapper-root mantine-NumberInput-root"
      style="width: 120px;"
    >
      <label
        class="m_8fdc1311 mantine-InputWrapper-label mantine-NumberInput-label"
        for="mantine-jb17163tz"
        id="mantine-jb17163tz-label"
      >
        Add/Remove $5
      </label>
      <div
        class="m_6c018570 mantine-Input-wrapper mantine-NumberInput-wrapper"
        data-variant="default"
        data-with-right-section="true"
        style="--input-right-section-width: var(--ni-right-section-width-sm);"
      >
        <input
          aria-invalid="false"
          class="m_8fb7ebe7 mantine-Input-input mantine-NumberInput-input"
          data-variant="default"
          id="mantine-jb17163tz"
          inputmode="numeric"
          type="text"
          value="1"
        />
        <div
          class="m_82577fc2 mantine-Input-section mantine-NumberInput-section"
          data-position="right"
        >
          <div
            class="m_95e17d22 mantine-NumberInput-controls"
          >
            <button
              aria-hidden="true"
              class="mantine-focus-auto m_80b4b171 mantine-NumberInput-control m_87cf2631 mantine-UnstyledButton-root"
              data-direction="up"
              tabindex="-1"
              type="button"
            >
              <svg
                fill="none"
                style="transform: rotate(180deg);"
                viewBox="0 0 15 15"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M3.13523 6.15803C3.3241 5.95657 3.64052 5.94637 3.84197 6.13523L7.5 9.56464L11.158 6.13523C11.3595 5.94637 11.6759 5.95657 11.8648 6.15803C12.0536 6.35949 12.0434 6.67591 11.842 6.86477L7.84197 10.6148C7.64964 10.7951 7.35036 10.7951 7.15803 10.6148L3.15803 6.86477C2.95657 6.67591 2.94637 6.35949 3.13523 6.15803Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
            </button>
            <button
              aria-hidden="true"
              class="mantine-focus-auto m_80b4b171 mantine-NumberInput-control m_87cf2631 mantine-UnstyledButton-root"
              data-direction="down"
              tabindex="-1"
              type="button"
            >
              <svg
                fill="none"
                viewBox="0 0 15 15"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M3.13523 6.15803C3.3241 5.95657 3.64052 5.94637 3.84197 6.13523L7.5 9.56464L11.158 6.13523C11.3595 5.94637 11.6759 5.95657 11.8648 6.15803C12.0536 6.35949 12.0434 6.67591 11.842 6.86477L7.84197 10.6148C7.64964 10.7951 7.35036 10.7951 7.15803 10.6148L3.15803 6.86477C2.95657 6.67591 2.94637 6.35949 3.13523 6.15803Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="m_e2f5cd4e m_46b77525 mantine-InputWrapper-root mantine-NumberInput-root"
      style="width: 120px;"
    >
      <label
        class="m_8fdc1311 mantine-InputWrapper-label mantine-NumberInput-label"
        for="mantine-0ug19yp05"
        id="mantine-0ug19yp05-label"
      >
        Add/Remove $10
      </label>
      <div
        class="m_6c018570 mantine-Input-wrapper mantine-NumberInput-wrapper"
        data-variant="default"
        data-with-right-section="true"
        style="--input-right-section-width: var(--ni-right-section-width-sm);"
      >
        <input
          aria-invalid="false"
          class="m_8fb7ebe7 mantine-Input-input mantine-NumberInput-input"
          data-variant="default"
          id="mantine-0ug19yp05"
          inputmode="numeric"
          type="text"
          value="1"
        />
        <div
          class="m_82577fc2 mantine-Input-section mantine-NumberInput-section"
          data-position="right"
        >
          <div
            class="m_95e17d22 mantine-NumberInput-controls"
          >
            <button
              aria-hidden="true"
              class="mantine-focus-auto m_80b4b171 mantine-NumberInput-control m_87cf2631 mantine-UnstyledButton-root"
              data-direction="up"
              tabindex="-1"
              type="button"
            >
              <svg
                fill="none"
                style="transform: rotate(180deg);"
                viewBox="0 0 15 15"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M3.13523 6.15803C3.3241 5.95657 3.64052 5.94637 3.84197 6.13523L7.5 9.56464L11.158 6.13523C11.3595 5.94637 11.6759 5.95657 11.8648 6.15803C12.0536 6.35949 12.0434 6.67591 11.842 6.86477L7.84197 10.6148C7.64964 10.7951 7.35036 10.7951 7.15803 10.6148L3.15803 6.86477C2.95657 6.67591 2.94637 6.35949 3.13523 6.15803Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
            </button>
            <button
              aria-hidden="true"
              class="mantine-focus-auto m_80b4b171 mantine-NumberInput-control m_87cf2631 mantine-UnstyledButton-root"
              data-direction="down"
              tabindex="-1"
              type="button"
            >
              <svg
                fill="none"
                viewBox="0 0 15 15"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M3.13523 6.15803C3.3241 5.95657 3.64052 5.94637 3.84197 6.13523L7.5 9.56464L11.158 6.13523C11.3595 5.94637 11.6759 5.95657 11.8648 6.15803C12.0536 6.35949 12.0434 6.67591 11.842 6.86477L7.84197 10.6148C7.64964 10.7951 7.35036 10.7951 7.15803 10.6148L3.15803 6.86477C2.95657 6.67591 2.94637 6.35949 3.13523 6.15803Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="m_e2f5cd4e m_46b77525 mantine-InputWrapper-root mantine-NumberInput-root"
      style="width: 120px;"
    >
      <label
        class="m_8fdc1311 mantine-InputWrapper-label mantine-NumberInput-label"
        for="mantine-2o07noige"
        id="mantine-2o07noige-label"
      >
        Add/Remove $20
      </label>
      <div
        class="m_6c018570 mantine-Input-wrapper mantine-NumberInput-wrapper"
        data-variant="default"
        data-with-right-section="true"
        style="--input-right-section-width: var(--ni-right-section-width-sm);"
      >
        <input
          aria-invalid="false"
          class="m_8fb7ebe7 mantine-Input-input mantine-NumberInput-input"
          data-variant="default"
          id="mantine-2o07noige"
          inputmode="numeric"
          type="text"
          value="1"
        />
        <div
          class="m_82577fc2 mantine-Input-section mantine-NumberInput-section"
          data-position="right"
        >
          <div
            class="m_95e17d22 mantine-NumberInput-controls"
          >
            <button
              aria-hidden="true"
              class="mantine-focus-auto m_80b4b171 mantine-NumberInput-control m_87cf2631 mantine-UnstyledButton-root"
              data-direction="up"
              tabindex="-1"
              type="button"
            >
              <svg
                fill="none"
                style="transform: rotate(180deg);"
                viewBox="0 0 15 15"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M3.13523 6.15803C3.3241 5.95657 3.64052 5.94637 3.84197 6.13523L7.5 9.56464L11.158 6.13523C11.3595 5.94637 11.6759 5.95657 11.8648 6.15803C12.0536 6.35949 12.0434 6.67591 11.842 6.86477L7.84197 10.6148C7.64964 10.7951 7.35036 10.7951 7.15803 10.6148L3.15803 6.86477C2.95657 6.67591 2.94637 6.35949 3.13523 6.15803Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
            </button>
            <button
              aria-hidden="true"
              class="mantine-focus-auto m_80b4b171 mantine-NumberInput-control m_87cf2631 mantine-UnstyledButton-root"
              data-direction="down"
              tabindex="-1"
              type="button"
            >
              <svg
                fill="none"
                viewBox="0 0 15 15"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M3.13523 6.15803C3.3241 5.95657 3.64052 5.94637 3.84197 6.13523L7.5 9.56464L11.158 6.13523C11.3595 5.94637 11.6759 5.95657 11.8648 6.15803C12.0536 6.35949 12.0434 6.67591 11.842 6.86477L7.84197 10.6148C7.64964 10.7951 7.35036 10.7951 7.15803 10.6148L3.15803 6.86477C2.95657 6.67591 2.94637 6.35949 3.13523 6.15803Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="m_e2f5cd4e m_46b77525 mantine-InputWrapper-root mantine-NumberInput-root"
      style="width: 120px;"
    >
      <label
        class="m_8fdc1311 mantine-InputWrapper-label mantine-NumberInput-label"
        for="mantine-dvp9pyash"
        id="mantine-dvp9pyash-label"
      >
        Add/Remove $50
      </label>
      <div
        class="m_6c018570 mantine-Input-wrapper mantine-NumberInput-wrapper"
        data-variant="default"
        data-with-right-section="true"
        style="--input-right-section-width: var(--ni-right-section-width-sm);"
      >
        <input
          aria-invalid="false"
          class="m_8fb7ebe7 mantine-Input-input mantine-NumberInput-input"
          data-variant="default"
          id="mantine-dvp9pyash"
          inputmode="numeric"
          type="text"
          value="1"
        />
        <div
          class="m_82577fc2 mantine-Input-section mantine-NumberInput-section"
          data-position="right"
        >
          <div
            class="m_95e17d22 mantine-NumberInput-controls"
          >
            <button
              aria-hidden="true"
              class="mantine-focus-auto m_80b4b171 mantine-NumberInput-control m_87cf2631 mantine-UnstyledButton-root"
              data-direction="up"
              tabindex="-1"
              type="button"
            >
              <svg
                fill="none"
                style="transform: rotate(180deg);"
                viewBox="0 0 15 15"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M3.13523 6.15803C3.3241 5.95657 3.64052 5.94637 3.84197 6.13523L7.5 9.56464L11.158 6.13523C11.3595 5.94637 11.6759 5.95657 11.8648 6.15803C12.0536 6.35949 12.0434 6.67591 11.842 6.86477L7.84197 10.6148C7.64964 10.7951 7.35036 10.7951 7.15803 10.6148L3.15803 6.86477C2.95657 6.67591 2.94637 6.35949 3.13523 6.15803Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
            </button>
            <button
              aria-hidden="true"
              class="mantine-focus-auto m_80b4b171 mantine-NumberInput-control m_87cf2631 mantine-UnstyledButton-root"
              data-direction="down"
              tabindex="-1"
              type="button"
            >
              <svg
                fill="none"
                viewBox="0 0 15 15"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M3.13523 6.15803C3.3241 5.95657 3.64052 5.94637 3.84197 6.13523L7.5 9.56464L11.158 6.13523C11.3595 5.94637 11.6759 5.95657 11.8648 6.15803C12.0536 6.35949 12.0434 6.67591 11.842 6.86477L7.84197 10.6148C7.64964 10.7951 7.35036 10.7951 7.15803 10.6148L3.15803 6.86477C2.95657 6.67591 2.94637 6.35949 3.13523 6.15803Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="m_e2f5cd4e m_46b77525 mantine-InputWrapper-root mantine-NumberInput-root"
      style="width: 120px;"
    >
      <label
        class="m_8fdc1311 mantine-InputWrapper-label mantine-NumberInput-label"
        for="mantine-owwxzwmtw"
        id="mantine-owwxzwmtw-label"
      >
        Add/Remove $100
      </label>
      <div
        class="m_6c018570 mantine-Input-wrapper mantine-NumberInput-wrapper"
        data-variant="default"
        data-with-right-section="true"
        style="--input-right-section-width: var(--ni-right-section-width-sm);"
      >
        <input
          aria-invalid="false"
          class="m_8fb7ebe7 mantine-Input-input mantine-NumberInput-input"
          data-variant="default"
          id="mantine-owwxzwmtw"
          inputmode="numeric"
          type="text"
          value="1"
        />
        <div
          class="m_82577fc2 mantine-Input-section mantine-NumberInput-section"
          data-position="right"
        >
          <div
            class="m_95e17d22 mantine-NumberInput-controls"
          >
            <button
              aria-hidden="true"
              class="mantine-focus-auto m_80b4b171 mantine-NumberInput-control m_87cf2631 mantine-UnstyledButton-root"
              data-direction="up"
              tabindex="-1"
              type="button"
            >
              <svg
                fill="none"
                style="transform: rotate(180deg);"
                viewBox="0 0 15 15"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M3.13523 6.15803C3.3241 5.95657 3.64052 5.94637 3.84197 6.13523L7.5 9.56464L11.158 6.13523C11.3595 5.94637 11.6759 5.95657 11.8648 6.15803C12.0536 6.35949 12.0434 6.67591 11.842 6.86477L7.84197 10.6148C7.64964 10.7951 7.35036 10.7951 7.15803 10.6148L3.15803 6.86477C2.95657 6.67591 2.94637 6.35949 3.13523 6.15803Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
            </button>
            <button
              aria-hidden="true"
              class="mantine-focus-auto m_80b4b171 mantine-NumberInput-control m_87cf2631 mantine-UnstyledButton-root"
              data-direction="down"
              tabindex="-1"
              type="button"
            >
              <svg
                fill="none"
                viewBox="0 0 15 15"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M3.13523 6.15803C3.3241 5.95657 3.64052 5.94637 3.84197 6.13523L7.5 9.56464L11.158 6.13523C11.3595 5.94637 11.6759 5.95657 11.8648 6.15803C12.0536 6.35949 12.0434 6.67591 11.842 6.86477L7.84197 10.6148C7.64964 10.7951 7.35036 10.7951 7.15803 10.6148L3.15803 6.86477C2.95657 6.67591 2.94637 6.35949 3.13523 6.15803Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
